sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -379,6 +381,33 @@ class Analyzer(catalog: Catalog,
     }
   }
 
+  object ResolveSortAlias extends Rule[LogicalPlan] {
```

Comment:
Because `ResolveSortReferences` is used to fix the case `In many dialects of SQL it is valid to sort by attributes that are not present in the SELECT * clause.`, and
```
SELECT sum(key)
FROM src
GROUP BY value
ORDER BY sum(key)
```
is not this case, but it's error.